src/btree/bt_curnext.c

```diff
@@ -124,9 +124,18 @@ __cursor_fix_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
         cbt->v = __bit_getv_recno(cbt->ref, cbt->recno, btree->bitcnt);
         cbt->iface.value.data = &cbt->v;
     } else {
-        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK) && upd->type != WT_UPDATE_TOMBSTONE)
-            return (__wt_value_return(cbt, upd));
-        cbt->iface.value.data = upd->data;
+        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK)) {
```

Comment:
Optionally can add a comment "If this update has been restored from the disk, it needs to be freed after copying it to the user cursor."

Reply:
Fixed